src/CurrencyNetwork.ts

```diff
@@ -76,21 +76,12 @@ export class CurrencyNetwork {
    */
   public async getDecimals (networkAddress: string, decimals?: number): Promise<number> {
     await this._checkAddresses([networkAddress])
-    const isNetwork = await this.isNetwork(networkAddress)
-    if (isNetwork) {
-      return Promise.resolve(
-        ((typeof decimals === 'number') && decimals) ||
-        // TODO replace with list of known currency network in clientlib
-        this._utils.fetchUrl<NetworkDetails>(`networks/${networkAddress}`)
-          .then(network => network.decimals)
-      )
-    } else {
-      if ((typeof decimals === 'number') && decimals) {
-        return decimals
-      } else {
-        return Promise.reject(`${networkAddress} is a token address. Decimals have to be explicit.`)
-      }
+    if (decimals && typeof decimals === 'number') {
+      return decimals
     }
+    // TODO replace with list of known currency network in clientlib
+    const network = await this._utils.fetchUrl<NetworkDetails>(`networks/${networkAddress}`)
```

Comment:
What will happen if the address is not a network? Before there was a meaningful error, can we have that back?

Reply:
For example if a 404 comes back, we can throw an error like before.

Reply:
The meaningful error is thrown here:
https://github.com/trustlines-network/clientlib/blob/3205002e761441689f843153709638e1f81a4204/src/Utils.ts#L52
So basically it returns the error message sent by the relay server. Shouldn't this suffice?

Reply:
No, because the user of this function should not care about how you find out that it is not a network. Then the abstraction is broken. If we would check it differently maybe by comparing it to an internal list, the error should stay the same.

Reply:
Ok got it. Will throw an error like before then